### PR TITLE
support  setting the system variable `elasticjob.preferred.network.ip`

### LINKED ISSUE #1919: Specify an ip when job instance application deployed on multi-ip server with different network interface
## Feature Request

**For English only**, other languages will not accept.

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot make decision by current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Is your feature request related to a problem?
yes, [https://github.com/apache/shardingsphere-elasticjob/issues/1881](url)

### Describe the feature you would like.
The solution provided -Delasticjob.preferred.network.interface=eth0 can't meet the requirement as below:
When application instances deployed on different server, and each server has different netwrok interfaces, but we have unified deployment script, then no way to specify the network interface for each application instance.
I guess we can have a better way, refer to spring cloud's choice(spring.cloud.inetutils.preferredNetworks=ip regular express), specify the prefix of ip address, or use regular express. What do you think?

### PR DESCRIPTION
**Pull request by sunkai-cai**
Fixes #1919

Changes proposed in this pull request:
- support  setting the system variable `elasticjob.preferred.network.ip`
- add tests

### CODE REVIEW

File: elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/env/IpUtils.java [hunk 1]
```diff
@@ -113,7 +116,8 @@ private static NetworkInterface getFirstNetworkInterface(final List<NetworkInter
         }
         return result;
     }
-    
+
+    // For testing.
```
Comment (TeslaCN) [inline elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/env/IpUtils.java]: What is `for testing`?

File: elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/env/IpUtils.java [hunk 2]
```diff
@@ -129,7 +133,18 @@ private static boolean ignoreNetworkInterface(final NetworkInterface networkInte
             return true;
         }
     }
-    
+
+    // For testing.
```
Comment (TeslaCN) [inline elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/env/IpUtils.java]: What is `for testing`?

File: elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/env/IpUtils.java [hunk 3]
```diff
@@ -95,14 +97,15 @@ private static NetworkInterface findNetworkInterface() {
         }
         return result;
     }
-    
+
+    // For testing.
```
Comment (TeslaCN) [inline elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/env/IpUtils.java]: What is `for testing`?

File: elasticjob-infra/elasticjob-infra-common/src/test/java/org/apache/shardingsphere/elasticjob/infra/env/IpUtilsTest.java [hunk 4]
```diff
@@ -21,18 +21,105 @@
...
+    @SneakyThrows
+    public void assertGetFirstNetworkInterface() {
+        InetAddress address1 = mock(Inet4Address.class);
+        when(address1.isLoopbackAddress()).thenReturn(false);
+        when(address1.isAnyLocalAddress()).thenReturn(false);
+        when(address1.isReachable(100)).thenReturn(true);
+        when(address1.getHostAddress()).thenReturn("10.10.0.1");
+        Vector<InetAddress> addresses1 = new Vector<>();
+        addresses1.add(address1);
+
```
Comment (TeslaCN) [inline elasticjob-infra/elasticjob-infra-common/src/test/java/org/apache/shardingsphere/elasticjob/infra/env/IpUtilsTest.java]: Remove redundant blank lines.

**Review by TeslaCN — CHANGES_REQUESTED** [review]

File: elasticjob-infra/elasticjob-infra-common/src/test/java/org/apache/shardingsphere/elasticjob/infra/env/IpUtilsTest.java (hunk 4 above)
Comment (sunkai-cai) [reply elasticjob-infra/elasticjob-infra-common/src/test/java/org/apache/shardingsphere/elasticjob/infra/env/IpUtilsTest.java]: fine

File: elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/env/IpUtils.java (hunk 3 above)
Comment (sunkai-cai) [reply elasticjob-infra/elasticjob-infra-common/src/main/java/org/apache/shardingsphere/elasticjob/infra/env/IpUtils.java]: `for testing`  note the private method has  junit test. I remove this note.

**Review by Technoboy- — APPROVED** [review]
LGTM.
Could you post another pr for document updates? @skaic

**Comment by sunkai-cai** [comment]
> LGTM.
> Could you post another pr for document updates? @skaic

Of course

**Pull request merged**